### PR TITLE
syncer/dbconn: add more retry error

### LINKED ISSUE #1825: v2.0.4-Transaction-level retry for errors such as TxnLockNotFound
RT, 
txnlocknotfound may appear frequently in tidb-server in some scenarios
When tidb acts as the downstream of dm and undertakes online real-time query services, the performance regression and synchronization delay caused by safe-mode have a greater impact on the business
Hope that this transaction can be retried in other ways to reduce the probability of delay

dm-worker log：
`
[error="[code=10006:class=database:scope=not-set:level=high], Message: execute statement failed: commit, RawCause: Error 8022: Error: KV error safe to retry Txn(Mvcc(TxnLockNotFound { start_ts: TimeStamp(425904341916582174), commit_ts: TimeStamp(425904342991372376)`

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Auto retry for ~`mysql.ErrInvalidConn`~ and tidb's `kv.RetryableErr` (by https://github.com/pingcap/tidb-tools/pull/452). 

close #1825

### What is changed and how it works?
- ~When sql client interrupt (e.g. manually killed by `kill tidb xxx`), go-mysql will return `mysql.ErrInvalidConn` and this error is safe to auto-retry build the connection~
- tidb's `kv.RetryableErr` (currently only caused by TxnLockNotFound) is retryable too.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

**Review by Ehco1996 — APPROVED** [review]

File: syncer/dbconn/db.go [hunk 1]
```diff
@@ -160,6 +160,8 @@ func (conn *DBConn) ExecuteSQLWithIgnore(tctx *tcontext.Context, ignoreError fun
 						log.ShortError(err))
 					return false
 				}
+				tctx.L().Warn("execute sql failed by connection error", zap.Int("retry", retryTime),
+					zap.Error(err))
```
Comment (Ehco1996) [inline syncer/dbconn/db.go:164]: can we add test for this case?

Comment (glorv) [reply syncer/dbconn/db.go:164]: Will add test case for this after we add retry for all other connections so we can safely kill all connection between dm-worker and tidb

File: pkg/retry/errors.go [hunk 2]
```diff
@@ -82,7 +83,7 @@ var (
 func IsConnectionError(err error) bool {
 	err = errors.Cause(err)
 	switch err {
-	case driver.ErrBadConn, tmysql.ErrBadConn, gmysql.ErrBadConn:
+	case driver.ErrBadConn, tmysql.ErrBadConn, gmysql.ErrBadConn, mysql.ErrInvalidConn:
```
Comment (lance6716) [inline pkg/retry/errors.go:86]: we may not retry the `ErrInvalidConn` when it came from `txn.Commit()`

Comment (glorv) [reply pkg/retry/errors.go:86]: In the situation, do the other three error can safely retry?

Comment (glorv) [reply pkg/retry/errors.go:86]: ok

Comment (GMHDBJD) [reply pkg/retry/errors.go:86]: Currently, we will ignore invalid connection error for some DDL. https://github.com/pingcap/dm/blob/fadd54342759286e7572bb100684fe48942c0ca7/syncer/error.go#L79

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**